VideoLocker/src/main/java/org/edx/mobile/module/registration/view/IRegistrationFieldView.java

```diff
@@ -66,14 +66,8 @@ else if (fieldType.equals(RegistrationFieldType.MULTI)) {
                 return new RegistrationSelectView(field, view);
             }
             else if (fieldType.equals(RegistrationFieldType.CHECKBOX)) {
-                if (field.getAgreement() != null) {
```

Comment:
A is tricky because it seems like we are trying to avoid touching the API (it implicated gthe web as well).
I could do B, but I don't know how much better that will make the code, this is already hack-ish as is.
C would change the UI, I'm not sure how people feel about it.

I am open to a discussion on the best way to move forward.